File: Resources/Code/Data-Structure/Search-Algorithms/Jump-Search/python/JumpSearch.py
# import math 

class Search:
  def __init__(self, array_of_numbers):
    self.array_of_numbers = array_of_numbers

  def jump_search(self, number_to_find):
    array_size=len(self.array_of_numbers)
    stepSize=int(array_size**(1/2)) # int(math.sqrt(array_size))
    start=0
    end=stepSize

    while end < array_size - 1 and array_of_numbers[end] <= number_to_find:
        start = end
        end = end + stepSize
        if end > array_size - 1:
            end = array_size - 1

    while end >= start:
        if array_of_numbers[end] == number_to_find:
            return True
        end-=1

    return False

array_of_numbers = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
search = Search(array_of_numbers)
number_to_find = int(input('What number do you want find?: '))
print(search.jump_search(number_to_find))

File: Resources/Code/Data-Structure/Search/Jump-Search/cpp/JumpSearch.cpp
#include <iostream>
#include <cmath>

using namespace std;

int main()
{
    int array_of_numbers[10] = { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 };
    int number_to_find = 0;
    
    cout<<"What number do you want find?: ";
    cin>>number_to_find;
    int array_size = sizeof(array_of_numbers)/sizeof(int) - 1;
    int blockSize = pow(array_size, 2), start = 0, end = blockSize;

    while (end < array_size - 1 && array_of_numbers[end] <= number_to_find) {
        start = end;
        end = end + blockSize;
        if (end > array_size - 1) {
            end = array_size - 1;
        }
    }
    
    for (int i = end; i >= start; i--) {
        if (array_of_numbers[end] == number_to_find) {
            cout<<"True";
            return 0;
        }
        end--;
    }
        
    cout<<"False";
    return 0;
}